Lib/test/test_clinic.py

```diff
@@ -2508,6 +2508,15 @@ def test_state_func_docstring_only_one_param_template(self):
         """
         self.expect_failure(block, err, lineno=7)
 
+    def test_kind_defining_class(self):
+        function = self.parse_function("""
+            module m
+            m.func
+                cls: defining_class
```

Comment:
This test seems strange to me. IMO, it should not be allowed to use the `defining_class` converter if it is a module level function; a class method should be required.

Reply:
I merged my PR https://github.com/python/cpython/pull/117896 which enhance `@defining_class` tests. There are now 3 functional tests, it's maybe enough?

Reply:
Yes, I think that is enough.

Reply:
I think I fixed my code.  I'll post a PR after checking how to disallow `defining_class` placed on a module input.